src/modules/Plans/views/details/components/SettingsSection/components/TransferNetworkDetailsItem.tsx

```diff
@@ -25,29 +25,27 @@ export const TransferNetworkDetailsItem: FC<PlanDetailsItemProps> = ({
     network for all migration plans. Otherwise, the pod network is used.`,
   );
 
-  const TransferNetworkToName = (network: V1beta1PlanSpecTransferNetwork) =>
+  const transferNetworkToName = (network: V1beta1PlanSpecTransferNetwork) =>
     network && `${network.namespace}/${network.name}`;
 
-  const content = TransferNetworkToName(resource?.spec?.transferNetwork);
+  const content = transferNetworkToName(resource?.spec?.transferNetwork ?? {});
 
   return (
     <DetailsItem
       title={t('Transfer Network')}
       content={content || <span className="text-muted">{t('Providers default')}</span>}
       helpContent={helpContent ?? defaultHelpContent}
       crumbs={['spec', 'transferNetwork ']}
-      onEdit={
-        canPatch &&
-        isPlanEditable(resource) &&
-        (() => {
+      onEdit={() => {
+        if (canPatch && isPlanEditable(resource)) {
           showModal(
             <EditPlanTransferNetwork
               resource={resource}
-              destinationProvider={destinationProvider}
+              destinationProvider={destinationProvider!}
```

Comment:
kind of strange that this component takes the entire `PlanDetailsItemProps` props even though it doesn't use some of them. I don't know if this is any better, but throwing this out here:

```
type TransferNetworkDetailsItemProps = Partial<
  Pick<PlanDetailsItemProps, 'canPatch' | 'helpContent'>
> & {
  resource: PlanDetailsItemProps['resource'];
  destinationProvider: NonNullable<PlanDetailsItemProps['destinationProvider']>;
};
```

Reply:
the part of plan details page refactoring will be in a separated PR since this is getting too big, will check about this when refactoring, Thanks! @jschuler

Reply:
Great catch, fixed. Thanks!